arcgis-ios-sdk-samples/Display information/Create symbol styles from web styles/CreateSymbolStylesFromWebStylesViewController.swift

```diff
@@ -0,0 +1,306 @@
+    /// - Parameter featureLayer: An `AGSFeatureLayer` object.
+    /// - Returns: An `AGSMap` object.
+    func makeMap(featureLayer: AGSFeatureLayer) -> AGSMap {
+        let map = AGSMap(basemapStyle: .arcGISLightGray)
+        map.referenceScale = 1e5
+        map.operationalLayers.add(featureLayer)
+        return map
+    }
+    
+    // MARK: UIViewController
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Add the source code button item to the right of navigation bar.
+        (navigationItem.rightBarButtonItem as? SourceCodeBarButtonItem)?.filenames = ["CreateSymbolStylesFromWebStylesViewController"]
+        
+        // Get the symbols from the symbol style hosted on an ArcGIS portal.
+        symbolsDataSource.getSymbols(symbolStyle: symbolStyle, categories: SymbolCategory.allCases) { [weak self] in
+            guard let self = self else { return }
+            self.featureLayer.renderer = self.symbolsDataSource.makeUniqueValueRenderer(fieldNames: ["cat2"])
+            self.legendBarButtonItem.isEnabled = true
+        }
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        mapScaleObservation = mapView.observe(\.mapScale, options: .initial) { [weak self] (_, _) in
+            DispatchQueue.main.async {
+                guard let self = self else { return }
+                self.featureLayer.scaleSymbols = self.mapView.mapScale >= 8e4
+            }
+        }
+    }
+    
+    override func viewDidDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
+        mapScaleObservation = nil
+    }
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        if segue.identifier == "LegendTableSegue",
+           let controller = segue.destination as? UITableViewController {
+            controller.presentationController?.delegate = self
+            controller.tableView.dataSource = symbolsDataSource
+            // Hold a weak reference to the table view to asynchronously reload
+            // image swatches for the symbols.
+            symbolsDataSource.tableView = controller.tableView
+        }
+    }
+    
+    func adaptivePresentationStyle(for controller: UIPresentationController, traitCollection: UITraitCollection) -> UIModalPresentationStyle {
+        // Ensure that the settings are shown in a popover on small displays.
+        return .none
+    }
+}
+
+// MARK: - SymbolsDataSource
+
+private class SymbolsDataSource: NSObject {
+    /// A weak reference to the table view which the data source itself connects.
+    weak var tableView: UITableView?
+    
+    /// The symbols created from a symbol style.
+    private var symbols = [(category: SymbolCategory, symbol: AGSSymbol)]() {
+        didSet {
+            // Keep the array sorted whenever it is modified.
+            symbols.sort { $0.category < $1.category }
```

Comment:
I would suggest that the data source should display the data that is given in the order in which it was given. Therefore I suggest doing the sorting before assigning to to `symbols`, not after. That way it is only mutated once, not twice.